src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -336,4 +325,24 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
             return file;
         };
     }
+
+    /**
+     * Extract the bucket from a gs path.
+     */
+    public static String extractBucketFromPath(final String path) {
+        if (!path.startsWith("gs://") && StringUtils.countMatches(path, ',') < 3) {
+            throw new PdaoException(String.format("Bad GS path %s", path));
+        }
+        // Given a string that looks like "gs://some bucket/pat1/path2" (format verified above), this will extract
+        // "some bucket" and return
+        return path.substring(5).split("/")[0];
```

Comment:
IMO a clearer way to write this would be `return path.split('/')[2]`. Using `substring(5)` is pretty opaque. Using `GS_PROTOCOL.length()` would let you get rid of the magic number.

Reply:
Yep, I was thinking this based on your prior comment about the constant